src/lib/KeyStore.js

```diff
@@ -102,26 +116,39 @@ class KeyStore {
      * @returns {Promise<void>}
      */
     async put(key, passphrase) {
-        const secret = !passphrase
-            ? key.secret
-            : await Nimiq.CryptoUtils.encryptOtpKdf(new Nimiq.SerialBuffer(key.secret), passphrase);
+        /** @type {Uint8Array} */
+        let secret;
+        if (passphrase) {
+            secret = await key.secret.exportEncrypted(passphrase);
+        } else {
+            const buf = new Nimiq.SerialBuffer(KeyStore.UNENCRYPTED_SECRET_SIZE);
+            const purposeId = key.secret.type === Nimiq.Secret.Type.ENTROPY
+                ? Nimiq.Entropy.PURPOSE_ID
+                : Nimiq.PrivateKey.PURPOSE_ID;
+
+            buf.writeUint32(purposeId);
+            key.secret.serialize(buf);
+            secret = buf.subarray(0, buf.byteLength);
```

Comment:
Seems odd, that there is no other way to get the `Uint8Array`.

Reply:
Well, `SeiralBuffer` _extends_ `Uint8Array`, so maybe we can even pass in the raw SerialBuffer, but not sure if that works.

Reply:
But `secret` is typed to be a Uint8Array, and storing a `Nimiq.SerialBuffer` into the IndexedDB might produce unknown results, it's better to convert to Uint8Array, and this is the only way to do that.

Reply:
Yes. I looked it up myself, too. Was more about me being surprised.